### PR TITLE
rewrite hl.agg.corr in python

### PR DESCRIPTION
**Pull request by catoverdrive**
The Pearson correlation combiner was basically just tracking sums and counts of the base expressions, and combining them at the end. I copied the results equation in python and rewrote it directly in terms of the sum and count aggregators.

### CODE REVIEW

File: hail/src/test/scala/is/hail/expr/ir/AggregatorsSuite.scala [hunk 1]
```diff
@@ -630,25 +628,6 @@ class AggregatorsSuite extends HailSuite {
       FastIndexedSeq(Call2(0, 1), Call2(1, 1), null))
   }
 
-  @Test def pearsonCorrelationAggregator() {
```
Comment (jigold) [inline hail/src/test/scala/is/hail/expr/ir/AggregatorsSuite.scala]: Is there a corresponding test in Python?

Comment (catoverdrive) [reply hail/src/test/scala/is/hail/expr/ir/AggregatorsSuite.scala]: There is a test in python that runs the aggregator over a smaller set (once) and checks against scipy. I can make it run more times against a larger set, perhaps?

Comment (jigold) [reply hail/src/test/scala/is/hail/expr/ir/AggregatorsSuite.scala]: It looks like there's no null values. Might be good to add a second test with null values.

```
    def test_agg_corr(self):
        ht = hl.utils.range_table(10)
        ht = ht.annotate(x=hl.rand_unif(-10, 10),
                         y=hl.rand_unif(-10, 10))
        c, xs, ys = ht.aggregate((hl.agg.corr(ht.x, ht.y), hl.agg.collect(ht.x), hl.agg.collect(ht.y)))

        scipy_corr, _ = pearsonr(xs, ys)
        self.assertAlmostEqual(c, scipy_corr)
```

File: hail/python/test/hail/expr/test_expr.py [hunk 2]
```diff
@@ -903,12 +903,17 @@ def test_aggregator_group_by_sorts_result(self):
...
+                x=hl.cond(hl.rand_bool(0.1), hl.null(hl.tfloat64), hl.rand_unif(-10, 10)),
+                y=hl.cond(hl.rand_bool(0.1), hl.null(hl.tfloat64), hl.rand_unif(-10, 10)))))
 
-        scipy_corr, _ = pearsonr(xs, ys)
-        self.assertAlmostEqual(c, scipy_corr)
+        results = ht.aggregate(hl.agg.array_agg(lambda test: (hl.agg.corr(test.x, test.y), hl.agg.collect((test.x, test.y))), ht.tests))
+
+        for corr, xy in results:
+            filtered = [(x, y) for x, y in xy if x is not None and y is not None]
+            scipy_corr, _ = pearsonr([xy[0] for xy in filtered], [xy[1] for xy in filtered])
```
Comment (jigold) [inline hail/python/test/hail/expr/test_expr.py]: I prefer this:

`pearsonr([x for x, _ in filtered], [y for _, y in filtered])`

**Pull request merged**